backend/timed/tracking/migrations/0020_report_search_vector_trigger.py

```diff
@@ -0,0 +1,38 @@
+from django.contrib.postgres.search import SearchVector
+from django.db import migrations
+
+
+def compute_search_vector(apps, schema_editor):
+    Report = apps.get_model("tracking", "Report")
+    Report.objects.update(search_vector=SearchVector("comment"))
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        (
+            "tracking",
+            "0019_remove_report_search_vector_idx_report_search_vector_and_more",
+        ),
+    ]
+
+    operations = [
+        migrations.RunSQL(
+            sql="""
+            CREATE TRIGGER search_vector_trigger
+            BEFORE INSERT OR UPDATE OF comment
+            ON tracking_report
+            FOR EACH ROW EXECUTE PROCEDURE
+            tsvector_update_trigger(
+                search_vector, 'pg_catalog.english', comment
+            );
+            UPDATE tracking_report SET search_vector = NULL;
+            """,
+            reverse_sql="""
+            DROP TRIGGER IF EXISTS search_vector_trigger
+            ON tracking_report;
+            """,
+        ),
+        migrations.RunPython(
+            compute_search_vector, reverse_code=migrations.RunPython.noop
+        ),
```

Comment:
I'd rather have this in Python code.

Move the trigger to either the `Report` model's `save()` method, or into a `pre_save()` signal handler.

While SQL triggers and procedures are appealing, they are an absolute pain to manage and maintain, especially since Django's migration system doesn't know about them.

Peformance-wise this will be slightly worse, but not actually a problem

Reply:
i put it in a `post_save` signal handler